script.js

```diff
@@ -1,4 +1,221 @@
+var win = 0;
+var loss = 0;
+var currentMode = "Waiting for username";
+var username = "";
+var lastroundWinner = "";
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  if (currentMode == "Waiting for username") {
+    username = input;
+    currentMode = "gameSelection";
+    return `Welcome to the game, ${username}. Please enter either 'RPS' to play normal Rock Paper Scissors, or enter 'Korean RPS' to play muk-jji-ppa. You can add 'Robot' in front of either choice to play AI vs AI versions.`;
+  }
+  if (currentMode == "gameSelection") {
+    if (input == "Korean RPS") {
+      currentMode = "koreanRPS";
+      return `You have selected Korean RPS - muk-jji-ppa, ${username}.`;
+    }
+    if (input == "RPS") {
+      currentMode = "normalRPS";
+      return `You have selected normal RPS, ${username}.`;
+    }
+    if (input == "Robot Korean RPS") {
+      currentMode = "compkoreanRPS";
+      return `You have selected Robot Korean RPS - muk-jji-ppa, ${username}.`;
+    }
+    if (input == "Robot RPS") {
+      currentMode = "compnormalRPS";
```

Comment:
the general variable naming format is uppercase for the first letter of each word and lowercase for the first letter of the first word. So for 'compnormalRPS' it should be 'compNormalRPS'